src/univariate/discrete/geometric.jl

```diff
@@ -124,13 +124,12 @@ end
 
 function mgf(d::Geometric, t::Real)
     p = succprob(d)
-    p / (expm1(-t) + p)
+    p / (1 - (1 - p) * exp(t))
 end
 
 function cf(d::Geometric, t::Real)
     p = succprob(d)
-    # replace with expm1 when complex version available
-    p / (exp(-t*im) - 1 + p)
+    p / (1 - (1 - p) * cis(t))
```

Comment:
For the same reason as above, maybe use
```suggestion
    p / (p - (1 - p) * expm1(t*im))
```